src/Listener.cpp

```diff
@@ -143,9 +142,38 @@ std::string Listener::ThreadID()  {
 bool Listener::NotificationReceived() {
    std::vector<std::string> dataFromQueue;
    mQueueReader->GetShot(getShotTimeout, dataFromQueue);
-   return MessageHasPayload(dataFromQueue);
+   bool notificationReceived = MessageHasPayload(dataFromQueue);
+   ClearMessages();
```

Comment:
```ClearMessages()``` resets the member std::vector<std::string> so that we can store the new payload in the same vector if needed.

It can probably be shuffled a bit in this function, but it definitely needs to go above the ```if (notificationReceived)```